source/annotations/helpers_test.go

```diff
@@ -0,0 +1,189 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package annotations
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/external-dns/endpoint"
+)
+
+func TestParseAnnotationFilter(t *testing.T) {
+	tests := []struct {
```

Comment:
Just FYI, so does not need to be changed: you can also omit the "tests" var and start with the for loop and have an inline struct definition. I think it looks like this (out of my head without parser or compiler):
```
for _, tt := range []struct{
   name string
   ...
  }{
    name: "myName",
    ...
  }, {
     name: "anotherName",
     ....
  }, {
     name: "lastName",
     ....
  }} {
    t.Run(tt.name, func(t *testing.T) {
      // body
    })
}
```

Reply:
I'm not going to touch it, not because against, mainly due to consistency. There is not a single test with this design at the moment
![Screenshot 2025-03-28 at 21 48 40](https://github.com/user-attachments/assets/2663ee50-fd99-4d5c-8267-1baaa95689ba)